src/ethereum_spec_tools/evm_tools/t8n/t8n_types.py

```diff
@@ -205,7 +203,11 @@ def __init__(self, t8n: Any, stdin: Optional[Dict] = None):
                 elif not value.startswith("0x"):
                     data[address][key] = "0x" + hex(int(value))
 
-        self.state = t8n.json_to_state(data)
+        state = t8n.json_to_state(data)
+        if t8n.fork_module == "dao_fork":
+            t8n.fork.apply_dao(state)
```

Comment:
Shouldn't we run `apply_fork` for every fork, and not just the DAO fork?

Reply:
The `t8n` inputs do not have enough details to re-create the entire chain which is why we decided to use the `apply_body` function as the point of entry for the tool. That is why I have also separated out the state changes in `dao_fork` into a different function `apply_dao` and used that instead.

Reply:
I think forks that modify the state directly (like the DAO Fork) should be looked upon as more of an exception than a rule. But we can discuss this.

Reply:
I have created an issue for this for now (see #740 ). Would like to implement the next few steps in the evm tools and re-visit this at a later point